tripy/tests/spec_verification/op_doc_guide.md

```diff
@@ -0,0 +1,29 @@
+# Introduction
+
+Spec verification is designed to ensure that the datatypes documented in the operator documentation are accurate.
+
+# How to Verify an Operation
+
+To run the verification program on an operation, add the decorator @dtype_info.dtype_info to the operation. The inputs to the decorator will help the verifier determine the constraints on the inputs and the datatypes to verify.
+
+There are four optional inputs for the decorator:
+
+ -  `dtype_variables`: This input must be a dictionary with the names of groups of variables as the keys and lists of datatypes as the values. Example: dtype_variables={"T": ["float32", "float16", "int8", "int32", "int64", "bool"], "T1": ["int32"]}. Any datatype not included will be tested to ensure it fails the test cases.
+ - `dtype_constraints`: This input assigns inputs and return parameters to variable groups. It must be a dictionary with parameter names as keys and variable group names as values. For assigning the return value, the key must be dtype_info.RETURN_VALUE. Example: dtype_constraints={"input": "T", "index": "T1", dtype_info.RETURN_VALUE: "T"}.. 
+ - `param_type_specification`: This parameter addresses situations where the type hint is not defined or linked to an internal object builder. It also allows the verifier to use a type other than the first option in a Union type hint. Example: `param_type_specification={"self": "tripy.Tensor"}`. Here is a list of type hints that are currently being used: `"tripy.Tensor", "tripy.Shape", Sequence[int], numbers.Number, int, "tripy.dtype", datatype.dtype, Tuple, List[Union["tripy.Tensor"]], "tripy.device", bool, float`
```

Comment:
Is `param_type_specification` actually needed? It seems like the one place its used already has a type hint.

Reply:
It's used in a lot of places. For example in double assigned functions such as `__mul__` and `__rmul__` where for `__rmul__` we need to change the type of self to a number. There are also other cases where you have a union, and you want to pick which one you want to use for verification. If I remove it I will also have to remove double assigned function support and flexibility with picking which type to use.

Reply:
Why do you need `param_type_specification` for aliased functions? The type annotations don't change. In the case where there's a union, we should prefer `tripy.Tensor` so we can set the dtype explicitly and otherwise just pick arbitrarily.

Reply:
Isn't that information implied by `default_constraints_all`?

Reply:
It doesn't currently do that, but I move the features of `param_type_specification` to `default_constraints_all`.